File: server/types/pin.go
package types

type IPin struct {
	Id          string   `json:"id" bson:"_id,omitempty"`
	GeoCode     IGeoCode `json:"geocode" bson:"geoCode" binding:"required"`
	Name        string   `json:"name" bson:"name" binding:"required"`
	StarRating  int      `json:"starRating" bson:"starRating" binding:"required"`
	Traits      []ITrait `json:"traits" bson:"traits" binding:"required"`
	Description string   `json:"description" bson:"description"`
}

File: src/types/pin.ts

export interface ILocation{
  latitude: number;
  longitude: number;
};



export enum ITrait {
  WheelchairAccessible = "wheelchairAccessible",
  MultilingualStaff = "multilingualStaff",
  SignLanguage = "signLanguage",
  BrailleMenu = "brailleMenu",
  ServicePetFriendly = "servicePetFriendly",
  CuttingServices = "cuttingServices"
}


export interface IPin {
  id: string;
  geocode: ILocation;
  name: string;
  starRating: number;
  traits: ITrait[];
  description: string;
};


export interface IPinSearchCriteria {
  name: string;
  starRating?: number;
  traits: ITrait[];
}

